src/arrayMethodJoin.js

```diff
@@ -4,8 +4,13 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    return String(this.length === 0
+      ? ''
+      : this.map(el => el === undefined || el === null
+        ? ''
+        : el)
+        .reduce((result, curent) => result + String(separator) + curent));
```

Comment:
suggestion: split code to multiple lines only when line almost reaches max-length (80 chars)
try to solve without `String()` because it's a bit hard to read now